src/streamvbyte_zigzag.c

```diff
@@ -2,7 +2,7 @@
 
 static inline
 uint32_t _zigzag_encode_32 (int32_t val) {
-	return (val + val) ^ (val >> 31);
```

Comment:
The signed `val + val` here is the issue wrt. overflow causing undefined behavior